packages/core/src/shared/logger/toolkitLogger.ts

```diff
@@ -113,22 +113,22 @@ export class ToolkitLogger extends BaseLogger implements vscode.Disposable {
     }
 
     /* Format the message with topic header */
-    private addTopicToMessage(message: string | Error): string | Error {
+    private addTopicToMessage(message: string | Error): string | object {
         /*We shouldn't print unknow before current logging calls are migrated
          * TODO: remove this once migration of current calls is completed
          */
         if (this.topic === 'unknown') {
             return message
         }
-        const topicPrefix = `${this.topic}: `
+        const topic = `${this.topic}: `
         if (typeof message === 'string') {
-            return topicPrefix + message
+            return topic + message
         } else if (message instanceof Error) {
-            /* Create a new Error object to avoid modifying the original */
-            const topicError = new Error(topicPrefix + message.message)
-            topicError.name = message.name
-            topicError.stack = message.stack
-            return topicError
+            const errorWithTopic = {
```

Comment:
If our error object has other fields like `name` and `code` those would be lost by the looks of it.

Instead see if you can do:
```
(errorWithTopic as any).topic = topic
```
as this will forcefully add the value in.

Then you would revert the `object` return type back in to error.

---

Separately if we drop the `topic` variable we can skip needing to trim the `: ` from this topic value.
So instead on line 125 do `${this.topic}: ${this.message}`
and then below you can just do 
```
(errorWithTopic as any).topic = this.topic
```

Reply:
No doing so would not lose the fields like `name` and `code`, see:
<img width="1056" alt="Screenshot 2024-09-18 at 11 23 23 AM" src="https://github.com/user-attachments/assets/e25fc94a-3dd5-4afa-8aa0-262a7e0ab9ae">

Reply:
If we did what I mentioned above I think that would be cleaner as it looks like a regular error (except with an additional `topic` field), `instanceof Error` would still work, and the output wouldn't have a nested `message` property.

Reply:
This might cause the topic field to not be printed first, which is not idea since we want the topic to be the "header"

Upon discussion we are creating a new class to handle the errors with topic identifier. That way we ensures the ordering and allows us todo `instanceof`